### PR TITLE
Enable automatic scraping of metrics when SPO is installed as a system workload on OCP

### PR DESCRIPTION
**Pull request by jhrozek**
#### What type of PR is this?
/kind bug
(sort of?)

#### What this PR does / why we need it:
On OCP, the SPO is installed into the `openshift-security-profiles` namespace
which means that it's considered a "system workload" and the setup to scrape
metrics via Prometheus must be adjusted a bit.

I tried to explain every change on its own in the individual commit messages,
but the tl;dr is that we need to allow the `prometheus-k8s` SA to access the
endpoints (both in the CR sense and the HTTP endpoints) in the `openshift-security-profiles`
namespace and we need to put a special CA path into the ServiceMonitor.

The OCP downstream manifest is not strictly needed, I can do some sed or
python post-processing as part of creating the downstream bundle, but
the more changes I get to upstream, the less brittle the post-processing.

#### Which issue(s) this PR fixes:
None

#### Does this PR have test?
N/A

Unfortunately this change requires OCP to test. I will tag our QE engineer
so we can get some testing, though.

#### Special notes for your reviewer:
I might point some inline.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
Code LGTM

/hold 
for QE

That’s a great feature, I never considered that we can make it work out of the box. :)

**Comment by jhrozek** [comment]
@xiaojiey would you mind testing the PR with upstream first to avoid back-and-forth in case something doesn't work after the PR is merged?

This is how I tested myself:
1. have a cluster prepared and log into the cluster with `oc login` as `kubeadmin`
2. check out the PR, e.g. with `gh pr checkout 1358 --force`
3. push the images into the cluster with `make push-openshift-dev`
4. update the bundle files to be OCP-specific with `make bundle BUNDLE_OPERATOR_MANIFEST=deploy/openshift-downstream.yaml`
5. make the CSV use the image that was pushed into the cluster earlier: `sed -i "s#gcr.io/k8s-staging-sp-operator/security-profiles-operator:latest.*#image-registry.openshift-image-registry.svc:5000/openshift/security-profiles-operator:latest#g" bundle/manifests/security-profiles-operator.clusterserviceversion.yaml`
6. create a bundle image: `make bundle-build BUNDLE_IMG=quay.io/jhrozek/security-profiles-operator-bundle:metrics-ocp`
7. push the bundle image: `podman push quay.io/jhrozek/security-profiles-operator-bundle:metrics-ocp`
8. create the catalog image: `make catalog-build BUNDLE_IMGS=quay.io/jhrozek/security-profiles-operator-bundle:metrics-ocp CATALOG_IMG=quay.io/jhrozek/security-profiles-operator-catalog:metrics-ocp`
9. push the catalog image: `podman push quay.io/jhrozek/security-profiles-operator-catalog:metrics-ocp`
10. Deploy the catalog. I used this manifest:
```
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: security-profiles-operator
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/jhrozek/security-profiles-operator-catalog:metrics-ocp
```
12. deploy SPO. Since this is upstream, we'll fake the `openshift-security-profiles` namespace in the manifests:
```
$ cat ocp-ns.yaml
---
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-security-profiles
  labels:
    openshift.io/cluster-monitoring: "true"
    pod-security.kubernetes.io/enforce: privileged
    pod-security.kubernetes.io/audit: privileged
    pod-security.kubernetes.io/warn: privileged
$ cat og-all-namespaces.yaml
---
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: spo-all-namespaces
  namespace: openshift-security-profiles
$ cat subscription.yaml
---
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: security-profiles-operator-sub
  namespace: openshift-security-profiles
spec:
  channel: stable
  name: security-profiles-operator
  source: security-profiles-operator
  sourceNamespace: openshift-marketplace
```
13. deploy: `cat ocp-ns.yaml og-all-namespaces.yaml subscription.yaml | oc create -f -`
14. pretty much instantly, you should be able to see the SPO endpoints in the Web console under Observe->Targets. The endpoints will be down until the SPO DS is fully up, but should be up from that point on. When the endpoints are up, you should also be able to search the metrics in the UI.

**Comment by jhrozek** [comment]
@xiaojiey - there is one more thing related to uninstall I'd like to fix but probably as a follow-up PR. Since the ServiceMonitor is not owned by the DS or the operator, if you install SPO, then uninstall, but leave the namespace around, the endpoints will still be monitored by Prometheus until the `ServiceMonitor` is removed. In the typical uninstall case, this doesn't matter because the namespace will be removed as well with everything in it, but I think we should make a follow-up and remove the `ServiceMonitor` object when SPO goes away.

File: installation-usage.md
Comment (jhrozek) [inline installation-usage.md:1047]: @sheriff-rh as mentioned in https://github.com/openshift/openshift-docs/pull/52698, once this PR is merged, we no longer have to document that the user workload monitoring is to be enabled.

Comment (sheriff-rh) [reply installation-usage.md:1047]: Thank you!

**Comment by xiaojiey** [comment]
@jhrozek The PR works as expected. Verification pass with index image quay.io/jhrozek/security-profiles-operator-catalog:metrics-ocp.
Steps: 
1. install SPO
2. Check SPO metrics on GUI: it displayed as expected. No alert raised.

**Comment by jhrozek** [comment]
> @jhrozek The PR works as expected. Verification pass with index image quay.io/jhrozek/security-profiles-operator-catalog:metrics-ocp. Steps:
> 
>     1. install SPO
> 
>     2. Check SPO metrics on GUI: it displayed as expected. No alert raised.

Great, thank you for testing!
/hold cancel

**Pull request merged**
